### PR TITLE
feat: add get project endpoint under v1-alpha

### PR DESCRIPTION
**Pull request by epipav**
## Summary

Adds the first route in the public API, `GET /v1-alpha/projects/{slug}`, and turns on the `/v1-alpha` version it lives under. Development callers use it to learn which values to pass in `repos` and `platform`.

```json
{
  "slug": "kubernetes",
  "name": "Kubernetes",
  "repositories": [{ "url": "https://github.com/kubernetes/kubernetes" }],
  "connectedPlatforms": ["github", "git"]
}
```

- `api/src/versions/registry.ts` and `api/src/versions/v1-alpha/index.ts`: register `/v1-alpha`. Its spec is served at `/v1-alpha/openapi.json` as `1.0.0-alpha`. The docs reference page still renders only `/v1`, so alpha routes stay unannounced until the closed-alpha allowlist lands.
- `api/src/versions/v1-alpha/projects.ts`: the route. It makes one `projects_list` call (`slug`, `details=true`), the same call as the Nuxt handler.
- `api/src/lib/errors.ts`: `NotFoundError` (404 `not_found`) and `UpstreamUnavailableError` (503 `upstream_unavailable`), in the same style as `InvalidDateRangeError`. It also takes `notFoundHandler` from #2224's `src/errors/not-found.ts`, which is deleted: each branch had created a home for error code without seeing the other, and one file gives the error envelope and the promotion-time `GoneError` a single place to land. The version-bump playbook now points there ([review on #2224](https://github.com/linuxfoundation/insights/pull/2224#discussion_r4060858538)).
- `api/tests/version-lifecycle.test.ts` (from #2224): the registry case pinned `/v1` as the only version. It now expects `['/v1', '/v1-alpha']`, neither with lifecycle metadata.

## Decisions worth a look

- `repositories` is a list of `{ url }` objects. The Overviews group will add repo fields here, and objects keep that additive.
- Platform keys drop the `-nango` suffix and any resulting duplicates, the same way the UI does. Every value is then one a caller can pass straight into `platform`.
- An unknown slug returns 404. The epic's "unknown slug returns empty data" rule fits the metric endpoints, where an empty series is a valid answer.
- Every Tinybird failure returns 503 `upstream_unavailable`, and the original error is logged. Without this, Fastify would pass Tinybird's own status to the caller, so a Tinybird 401 would read as the caller's auth failure.
- One `onRequest` hook in the alpha plugin sets `Cache-Control: private, max-age=0` on every `/v1-alpha` response, errors included (ADR-0013). `api/` has no Redis response cache yet. When it lands, this endpoint belongs in the 24h tier.
- Error bodies use Fastify's default shape, with `code` set, until the error envelope lands.

## Testing

- `api/tests/project-endpoint.test.ts`: runs the real app and Tinybird client, with `fetch` stubbed at the HTTP boundary. It covers the response body, the pipe call, platform and repo edge cases, the 404, Tinybird 500/401/429, network and non-JSON failures mapping to 503, the cache header on successes and errors, and the `/v1-alpha` spec (version, `Projects` tag, field descriptions, kept out of `/v1`).
- `api/tests/openapi-versions.test.ts`: the registry now lists `['/v1', '/v1-alpha']`.

## References

- JIRA: IN-1345, a sub-task of IN-1141 (Endpoints phase 1: Development)
- \#2224 (IN-1139) and #2230 (IN-1330, adds `ProjectSlugParams`) are on `main` and merged into this branch. That merge is what let `notFoundHandler` move into `src/lib/errors.ts` and made the registry test expect `/v1-alpha`.

* `main` 
  - **feat: add get project endpoint under v1-alpha** :point\_left:
    - \#2236
    - \#2237
    - \#2238
    - \#2239
    - \#2240

### CODE REVIEW

File: api/src/versions/v1-alpha/index.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import projectRoutes from './projects.js';
+
+// Owns everything under /v1-alpha; the spec route is mounted by the registry loop in app.ts.
+const v1AlphaRoutes: FastifyPluginAsyncTypebox = async (scope) => {
+  // Set before the handler runs, so error replies carry it too (see ADR-0013).
+  scope.addHook('onRequest', async (_request, reply) => {
+    reply.header('Cache-Control', 'private, max-age=0');
+  });
+
+  await scope.register(projectRoutes);
```
[the remark this answers is not in the document]
Comment (epipav) [reply api/src/versions/v1-alpha/index.ts:13]: Out of scope of this pr: auth (T-017) and the closed-alpha allowlist (T-089) are separate tickets, and `/v1-alpha` stays unannounced until then. The api is not deployed yet, T-017 ships before it is.

File: api/src/versions/v1-alpha/index.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// Copyright (c) 2025 The Linux Foundation and each contributor.
+// SPDX-License-Identifier: MIT
+import type { FastifyPluginAsyncTypebox } from '@fastify/type-provider-typebox';
+import projectRoutes from './projects.js';
+
+// Owns everything under /v1-alpha; the spec route is mounted by the registry loop in app.ts.
+const v1AlphaRoutes: FastifyPluginAsyncTypebox = async (scope) => {
+  // Set before the handler runs, so error replies carry it too (see ADR-0013).
+  scope.addHook('onRequest', async (_request, reply) => {
+    reply.header('Cache-Control', 'private, max-age=0');
```
[the remark this answers is not in the document]
Comment (epipav) [reply api/src/versions/v1-alpha/index.ts:10]: Pre-existing: both spec routes come from the registry loop in `app.ts`, and `/v1/openapi.json` has the same gap. Setting the header in that loop for every version is a separate follow-up.

File: api/src/versions/v1-alpha/projects.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    async (request) => {
+      const { slug } = request.params;
+      const { data } = await getTinybirdClient()
+        .fetch<ProjectRow[]>('/v0/pipes/projects_list.json', { slug, details: true })
+        .catch((err: unknown) => {
+          request.log.error({ err }, 'Tinybird projects_list request failed');
+          throw new UpstreamUnavailableError();
+        });
+
+      const [row] = data;
```
[the remark this answers is not in the document]
Comment (epipav) [reply api/src/versions/v1-alpha/projects.ts:55]: Pipe responses always carry `data` as an array, so a broken shape is a contract failure and a 500 fits better than a retryable 503. An array guard belongs in the shared client's `parseResponse` as a follow-up.

**Review by gaspergrom — APPROVED** [review]
Traced this against the Nuxt handler (frontend/server/api/project/[slug]/index.ts): same projects_list call, same -nango suffix stripping. Confirmed the deleted not-found.ts has no leftover references and the registry tests match the new /v1-alpha entry. Solid edge case coverage on the error mapping. LGTM.
